Walkthroughs/UserCommand/README.md

```diff
@@ -0,0 +1,214 @@
+# Understanding the USER Command
+### This walkthrough examines the USER command, provides best practices, and demonstrates how it works on a host
+
+## Overview 
+The `USER` and `GROUP` commands in Docker are used to execute commands inside a container as a non-root user. If no `USER` or `GROUP` is provided, the container runs as root. This document provides some best practices and things to avoid, and ends with a guided demo for the user to walk through the use of USER/GROUP to better explain the rationale behind the tips given.
+
+### Best Practices
+
+1. **Use a Non-root User**: 
+    - For security reasons, it's recommended to run processes inside a container as a non-root user. If an attacker manages to break out of the application running in the container, they will have limited permissions if the container is running as a non-root user.
+
+2. **Specify User by UID and GID**:
+    - Instead of specifying a user by name, use the user's UID and GID. This ensures that the user is consistently identified, even if the container's `/etc/passwd` file changes or is different across distributions.
+      ```Dockerfile
+      USER 1001:1001
+      ```
```

Comment:
Linking off to an explanation of Linux UIDs could be helpful here.

Reply:
How does this work on Windows?

Reply:
In the Linux VM it works the same - in Windows containers it seems to do similar (in that you can assign anything). I need to dig more on that side, so I'll put a note about this being linux containers.